Meadowlark-js/packages/meadowlark-core/test/validation/ReferenceAndIdentityEquivalence.test.ts

```diff
@@ -0,0 +1,336 @@
+          classPeriodName: 'z2',
+        },
+      },
+    ],
+  };
+
+  const classPeriodIdentityBody = {
+    classPeriodName: 'z1',
+    schoolReference: {
+      schoolId: '24',
+    },
+  };
+
+  const courseOfferingIdentityBody = {
+    localCourseCode: 'abc',
+    schoolReference: {
+      schoolId: '23',
+    },
+  };
+
+  beforeAll(() => {
+    MetaEdTextBuilder.build()
+      .withBeginNamespace('EdFi')
+      .withStartDomainEntity('Section')
+      .withDocumentation('doc')
+      .withStringIdentity('SectionIdentifier', 'doc', '30')
+      .withDomainEntityIdentity('CourseOffering', 'doc')
+      .withDomainEntityProperty('ClassPeriod', 'doc', true, true)
+      .withEndDomainEntity()
+
+      .withStartDomainEntity('CourseOffering')
+      .withDocumentation('doc')
+      .withStringIdentity('LocalCourseCode', 'doc', '30')
+      .withDomainEntityIdentity('School', 'doc')
+      .withEndDomainEntity()
+
+      .withStartDomainEntity('ClassPeriod')
+      .withDocumentation('doc')
+      .withStringIdentity('ClassPeriodName', 'doc', '30')
+      .withDomainEntityIdentity('School', 'doc')
+      .withEndDomainEntity()
+
+      .withStartDomainEntity('School')
+      .withDocumentation('doc')
+      .withStringIdentity('SchoolId', 'doc', '30')
+      .withEndDomainEntity()
+      .withEndNamespace()
+      .sendToListener(new NamespaceBuilder(metaEd, []))
+      .sendToListener(new DomainEntityBuilder(metaEd, []));
+
+    domainEntityReferenceEnhancer(metaEd);
+    entityPropertyMeadowlarkDataSetupEnhancer(metaEd);
+    entityMeadowlarkDataSetupEnhancer(metaEd);
+    referenceComponentEnhancer(metaEd);
+    apiPropertyMappingEnhancer(metaEd);
+    propertyCollectingEnhancer(metaEd);
+    apiEntityMappingEnhancer(metaEd);
+
+    namespace = metaEd.namespace.get('EdFi');
+    const section = namespace.entity.domainEntity.get('Section');
+    referenceExtractionResult = extractDocumentReferences(section, bodyWithReferences);
+    const classPeriod = namespace.entity.domainEntity.get('ClassPeriod');
+    classPeriodIdentityExtractionResult = extractDocumentIdentity(classPeriod, classPeriodIdentityBody).documentIdentity;
+    const courseOffering = namespace.entity.domainEntity.get('CourseOffering');
+    courseOfferingIdentityExtractionResult = extractDocumentIdentity(
+      courseOffering,
+      courseOfferingIdentityBody,
+    ).documentIdentity;
+  });
+
+  it('should have classPeriod reference and identity match', () => {
+    const [classPeriodReferenceExtractonResult] = referenceExtractionResult;
+    expect(classPeriodReferenceExtractonResult.documentIdentity.sort(valueCompare)).toEqual(
+      classPeriodIdentityExtractionResult.sort(valueCompare),
+    );
+  });
+
+  it('should have courseOffering reference and identity match', () => {
+    const [, , courseOfferingReferenceExtractonResult] = referenceExtractionResult;
+    expect(courseOfferingReferenceExtractonResult.documentIdentity.sort(valueCompare)).toEqual(
```

Comment:
Sort before testing because both identity and reference results are arrays where order is not important